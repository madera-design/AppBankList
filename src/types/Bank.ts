export type Bank = {
    bankName: string;
    description: string;
    age: number;
    url: string;
  };