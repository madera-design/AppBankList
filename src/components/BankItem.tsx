import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import { Bank } from '../types/Bank';

const BankItem = ({ bank }: { bank: Bank })  => {
  return (
    <View style={styles.card}>
      <Image 
        testID="bank-image" 
        source={{ uri: bank.url }} 
        style={styles.image} 
        resizeMode="contain"
       />
      <View style={styles.textContainer}>
        <Text style={styles.title}>{bank.bankName}</Text>
        <Text style={styles.description}>{bank.description}</Text>
        <Text style={styles.age}>AÃ±os en servicio: {bank.age}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#ffffff',
    padding: 15,
    marginVertical: 8,
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 6,
    elevation: 3,
    flexDirection: 'row',
    alignItems: 'center',
  },
  textContainer: {
    flex: 1,
    marginLeft: 15,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#333',
  },
  description: {
    fontSize: 14,
    color: '#555',
    marginTop: 4,
  },
  age: {
    fontSize: 12,
    fontStyle: 'italic',
    color: '#777',
    marginTop: 6,
  },
  image: {
    width: 60,
    height: 60,
    borderRadius: 30,
    backgroundColor: '#eee',
  },
});

export default BankItem;
