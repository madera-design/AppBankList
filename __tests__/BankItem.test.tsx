import React from 'react';
import { render } from '@testing-library/react-native';
import BankItem from '../src/components/BankItem';
import { Bank } from '../src/types/Bank';

describe('BankItem Component', () => {
  const mockBank: Bank = {
    bankName: 'Banco de Prueba',
    description: 'Descripción de prueba',
    age: 10,
    url: 'https://example.com/logo.png',
  };

  it('renders correctly with bank data', () => {
    const { getByText, getByTestId } = render(<BankItem bank={mockBank} />);

    // Verifica que el nombre del banco se renderice
    expect(getByText('Banco de Prueba')).toBeTruthy();

    // Verifica que la descripción se renderice
    expect(getByText('Descripción de prueba')).toBeTruthy();

    // Verifica que los años en servicio se rendericen
    expect(getByText('Años en servicio: 10')).toBeTruthy();

    // Verifica que la imagen se renderice con la URL correcta
    const image = getByTestId('bank-image');
    expect(image.props.source.uri).toBe('https://example.com/logo.png');
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<BankItem bank={mockBank} />);
    expect(toJSON()).toMatchSnapshot();
  });
});