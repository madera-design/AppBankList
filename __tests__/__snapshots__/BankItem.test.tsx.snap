// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankItem Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "elevation": 3,
      "flexDirection": "row",
      "marginVertical": 8,
      "padding": 15,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 6,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://example.com/logo.png",
      }
    }
    style={
      {
        "backgroundColor": "#eee",
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
    testID="bank-image"
  />
  <View
    style={
      {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Banco de Prueba
    </Text>
    <Text
      style={
        {
          "color": "#555",
          "fontSize": 14,
          "marginTop": 4,
        }
      }
    >
      Descripción de prueba
    </Text>
    <Text
      style={
        {
          "color": "#777",
          "fontSize": 12,
          "fontStyle": "italic",
          "marginTop": 6,
        }
      }
    >
      Años en servicio: 
      10
    </Text>
  </View>
</View>
`;
